test/trace.ts

```diff
@@ -61,9 +61,11 @@ class TestTraceWriter extends TraceWriter {
     });
   }
 }
+setTraceWriterEnabled(false);
 
-let singleton: TraceWriter|null = null;
-disableTraceWriter();
+export function setTraceWriterEnabled(enabled: boolean) {
+  traceWriter['implementation'] = enabled ? TraceWriter : TestTraceWriter;
```

Comment:
why not `traceWriter.implementation`?

Reply:
`traceWriter.implementation` is a private name. TypeScript allows `traceWriter['implementation']` as a work-around for that.